sync-core/src/test/java/com/cloudant/sync/query/QueryExecutorTest.java

```diff
@@ -0,0 +1,283 @@
+        bodyMap.put("pet", "cat");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "mike34";
+        bodyMap.clear();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 34);
+        bodyMap.put("pet", "dog");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "mike72";
+        bodyMap.clear();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 72);
+        bodyMap.put("pet", "cat");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "fred34";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 34);
+        bodyMap.put("pet", "cat");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "fred12";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 12);
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        List<Object> fieldNames = new ArrayList<Object>(Arrays.asList("name", "age"));
+        Assert.assertEquals("basic", im.ensureIndexed(fieldNames, "basic"));
+
+        fieldNames = new ArrayList<Object>(Arrays.asList("name", "pet"));
+        Assert.assertEquals("pet", im.ensureIndexed(fieldNames, "pet"));
+    }
+
+    @Test
+    public void returnsNullForNoQuery() {
+        Assert.assertNull(im.find(null));
+    }
+
+    @Test
+    public void canQueryOverOneStringField() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query);
+        Assert.assertNotNull(queryResult);
+        Assert.assertEquals(3, queryResult.size());
+        List<String> checkList = Arrays.asList("mike12", "mike34", "mike72");
+        Assert.assertEquals(3, queryResult.documentIds().size());
+        Assert.assertTrue(queryResult.documentIds().containsAll(checkList));
```

Comment:
Done in f9ecc68.  Included in PR:  https://github.com/cloudant/sync-android/pull/80